### ISSUE TITLE
magic.solana.signMessage fails on react-native

### ISSUE THREAD

**Issue by kelly-ry4n** [magiclabs/magic-js#355]
### ✅ Prerequisites

- [ ✅ ] Did you perform a cursory search of open issues? Is this bug already reported elsewhere?

Mentioned in https://github.com/magiclabs/magic-js/issues/354 but closed - I do not think the issue is resolved.

- [ ✅ ] Are you running the latest SDK version?
- [ ✅ ] Are you reporting to the correct repository (`magic-sdk`)?

### 🐛 Description

The web codesandbox here https://codesandbox.io/s/ma-otp-forked-8yt5e7 works but copied line-for-line into react native, and changing

```diff
-import { Magic } from "magic-sdk";
+import { Magic } from "magic-sdk/react-native

-    await magicClient.auth.loginWithMagicLink({
-      email: email,
-    });

+    await magicClient.auth.loginWithSms({
+      phoneNumber: phoneNumber,
+    });
```

Seems to break it. 

The Magic client instantiates,  Logging in with magic works, but the `signMessage` method throws that error no matter what I provide as an argument.

For example:

```javascript
      const bytes = new Uint8Array([72, 101, 108, 108, 111]); // "Hello"
      const signature = await magicClient.solana.signMessage(bytes);
```

throws

```
 [Error: Magic RPC Error: [-32603] Internal error: unexpected type, use Uint8Array]
```

Do you have an example of this working in a react-native environment?

### 🤔 Expected behavior

`signMessage` returns a signed message

### 😮 Actual behavior

Throws

```
 [Error: Magic RPC Error: [-32603] Internal error: unexpected type, use Uint8Array]
```

### 🌎 Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| `magic-sdk/react-native`      | 10.0.0
| ios          | 15.5
| iPhone          | 13 pro
| `npm`           | 8.12.2
| react-native | >=0.68.1

**Comment by Ethella** [comment]
Hey @kelly-ry4n. Thx for reporting this. I'm taking a look now and shall get back to you shortly.

**Comment by Ethella** [comment]
@kelly-ry4n, the issue is caused by `JSON.stringify` mangling the original object structure. I'll make a patch soon, and it may requires you to update the Magic RN SDK

**Comment by vivianfastaf** [comment]
Thanks @Ethella -- Do you have an ETA on the patch fix?

**Comment by Ethella** [comment]
> Thanks @Ethella -- Do you have an ETA on the patch fix? 

It should be out today. I will leave a note in the thread.

**Comment by magiclabsFE** [comment]
:rocket: Issue was released in [`@magic-sdk/react-native@10.0.1`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/react-native%4010.0.1) :rocket:
